cmd/registry/cmd/diff/diff.go

```diff
@@ -0,0 +1,210 @@
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/log"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/iterator"
+)
+
+func Command(ctx context.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "diff",
+		Short: "Compare resources in the registry",
+		Args:  cobra.MinimumNArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			client, err := connection.NewClient(ctx)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get client")
+			}
+
+			var spec1, spec2 *rpc.ApiSpec
+			var path1 string
+			if name1, err := names.ParseSpec(args[0]); err == nil {
+				err = core.GetSpec(ctx, client, name1, true, func(s *rpc.ApiSpec) error {
+					spec1 = s
+					return nil
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to compare resources")
+				}
+				path1 = name1.String()
+			} else if name1, err := names.ParseSpecRevision(args[0]); err == nil {
+				err = core.GetSpecRevision(ctx, client, name1, true, func(s *rpc.ApiSpec) error {
+					spec1 = s
+					return nil
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to compare resources")
+				}
+				path1 = name1.Spec().String()
+			}
+
+			if name2, err := names.ParseSpec(args[1]); err == nil {
+				err = core.GetSpec(ctx, client, name2, true, func(s *rpc.ApiSpec) error {
+					spec2 = s
+					return nil
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to compare resources")
+				}
+			} else if name2, err := names.ParseSpecRevision(args[1]); err == nil {
+				err = core.GetSpecRevision(ctx, client, name2, true, func(s *rpc.ApiSpec) error {
+					spec2 = s
+					return nil
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to compare resources")
+				}
+			} else if name2, err := resolveSpecRevision(ctx, client, path1, args[1]); err == nil {
+				err = core.GetSpecRevision(ctx, client, name2, true, func(s *rpc.ApiSpec) error {
+					spec2 = s
+					return nil
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to compare resources")
+				}
+			} else {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to match or handle command")
+			}
+			if spec1 != nil && spec2 != nil {
+				err = printDiff(spec1, spec2)
+			}
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to match or handle command")
+			}
+		},
+	}
+	return cmd
+}
+
+func resolveSpecRevision(ctx context.Context,
+	client connection.Client,
+	base string,
+	suffix string) (names.SpecRevision, error) {
+	// First try to treat the raw suffix as revision name.
+	name, err := names.ParseSpecRevision(fmt.Sprintf("%s%s", base, suffix))
```

Comment:
Agreed and frankly, I disagree with the push to use fmt.Sprintf() in place of string concatenation because it is much slower, but I've been doing it because it came up as readability guidance.